src/devices/Camera/Camera/Settings/CommandOptionsBuilder.cs

```diff
@@ -0,0 +1,393 @@
+    /// <param name="width">The width of the image.</param>
+    /// <param name="height">The heighy of the image.</param>
+    /// <returns>A reference to this instance.</returns>
+    public CommandOptionsBuilder WithResolution(int width, int height)
+    {
+        AddWidthAndHeight(width, height);
+        return this;
+    }
+
+    /// <summary>
+    /// Adds the image quality options.
+    /// </summary>
+    /// <param name="sharpness">The sharpness value (must be positive, typically no more than 2.0).</param>
+    /// <param name="contrast">The contrast value (must be zero or positive, typically no more than 2.0).</param>
+    /// <param name="brightness">The brightness value (must be between -1.0=black and 1.0=white).</param>
+    /// <param name="saturation">The saturation value (0 is gray, 1.0 is the default, larger values are saturated).</param>
+    /// <returns>A reference to this instance.</returns>
+    public CommandOptionsBuilder WithImageQuality(decimal sharpness = 1.0m, decimal contrast = 1.0m,
+        decimal brightness = 0.0m, decimal saturation = 1.0m)
+    {
+        AddSharpness(sharpness);
+        AddContrast(contrast);
+        AddBrightness(brightness);
+        AddSaturation(saturation);
+        return this;
+    }
+
+    /// <summary>
+    /// Adds the option to take a still picture.
+    /// </summary>
+    /// <param name="quality">0 to 100, default is 93.</param>
+    /// <param name="encoding">One of the following strings: jpg, png, bmp, rgb, yuv420.</param>
+    /// <returns>A reference to this instance.</returns>
+    public CommandOptionsBuilder WithPictureOptions(int quality = 93, string encoding = "jpg")
+    {
+        AddPictureQuality(quality);
+        AddEncoding(encoding);
+        return this;
+    }
+
+    /// <summary>
+    /// Adds the option to capture a video stream in H.264 format.
+    /// This method will automatically set H264 and the 'inline' option that writes the H264 header to
+    /// every Intra frame. The frequency of Intra frames can be changed.
+    /// </summary>
+    /// <param name="profile">The H264 profile used by the hardware encoder: baseline, main or high.</param>
+    /// <param name="level">The level of the H264 protocol: 4, 4.1 or 4.2.</param>
+    /// <param name="intra">The frequency of I (Intra) frames (number of frames).</param>
+    /// <returns>A reference to this instance.</returns>
+    public CommandOptionsBuilder WithH264VideoOptions(string profile, string level, int intra = 60)
+    {
+        AddCodec("h264");
+        AddInline();
+        AddProfile(profile);
+        AddLevel(level);
+        AddIntra(intra);
+        return this;
+    }
+
+    /// <summary>
+    /// Adds the option to capture a video stream in MJPEG format.
+    /// </summary>
+    /// <param name="quality">The quality of each MJPEG picture.</param>
+    /// <returns>A reference to this instance.</returns>
+    public CommandOptionsBuilder WithMJPEGVideoOptions(int quality)
```

Comment:
nit: we don't do fluent APIs here elsewhere (I think), feel free to ignore this comment

Reply:
sure, I did it just for uniformity ... also it felt easier when using it in the sample/test code